studio/components/interfaces/Database/Schemas/SchemaGraph.tsx

```diff
@@ -0,0 +1,341 @@
+      tables,
+      rel.target_table_name,
+      rel.target_column_name
+    )
+
+    // We do not support [external->this] flow currently.
+    if (source && target) {
+      edges.push({
+        id: String(rel.id),
+        source,
+        sourceHandle,
+        target,
+        targetHandle,
+      })
+    }
+  }
+
+  return getLayoutedElements(nodes, edges)
+}
+
+function findTablesHandleIds(
+  tables: PostgresTable[],
+  table_name: string,
+  column_name: string
+): [string?, string?] {
+  for (const table of tables) {
+    if (table_name !== table.name) continue
+
+    for (const column of table.columns || []) {
+      if (column_name !== column.name) continue
+
+      return [String(table.id), column.id]
+    }
+  }
+
+  return []
+}
+
+const getLayoutedElements = (nodes: Node[], edges: Edge[]) => {
+  const dagreGraph = new dagre.graphlib.Graph()
+  dagreGraph.setDefaultEdgeLabel(() => ({}))
+  dagreGraph.setGraph({
+    rankdir: 'LR',
+    align: 'UR',
+    nodesep: 25,
+    ranksep: 50,
+  })
+
+  nodes.forEach((node) => {
+    dagreGraph.setNode(node.id, {
+      width: NODE_WIDTH / 2,
+      height: (NODE_ROW_HEIGHT / 2) * (node.data.columns.length + 1), // columns + header
+    })
+  })
+
+  edges.forEach((edge) => {
+    dagreGraph.setEdge(edge.source, edge.target)
+  })
+
+  dagre.layout(dagreGraph)
+
+  nodes.forEach((node) => {
+    const nodeWithPosition = dagreGraph.node(node.id)
+    node.targetPosition = Position.Left
+    node.sourcePosition = Position.Right
+    // We are shifting the dagre node position (anchor=center center) to the top left
+    // so it matches the React Flow node anchor point (top left).
+    node.position = {
+      x: nodeWithPosition.x - nodeWithPosition.width / 2,
+      y: nodeWithPosition.y - nodeWithPosition.height / 2,
+    }
+
+    return node
+  })
+
+  return { nodes, edges }
+}
+
+// ReactFlow is scaling everything by the factor of 2
+const NODE_WIDTH = 320
+const NODE_ROW_HEIGHT = 40
+
+function TableNode({ data, targetPosition, sourcePosition }: NodeProps<TableNodeData>) {
+  // Important styles is a nasty hack to use Handles (required for edges calculations), but do not show them in the UI.
+  // ref: https://github.com/wbkd/react-flow/discussions/2698
+  const hiddenNodeConnector = '!h-px !w-px !min-w-0 !min-h-0 !cursor-grab !border-0 !opacity-0'
+
+  return (
+    <>
+      {data.isForeign ? (
+        <div className="rounded-lg overflow-hidden">
+          <header className="text-[0.5rem] leading-5 font-bold px-2 text-center bg-yellow-900 text-gray-300">
+            {data.name}
+            {targetPosition && (
+              <Handle
+                type="target"
+                id={data.name}
+                position={targetPosition}
+                className={`${hiddenNodeConnector} !left-0`}
+              />
+            )}
+          </header>
+        </div>
+      ) : (
+        <div className={`w-[${NODE_WIDTH / 2}px] rounded-lg overflow-hidden`}>
```

Comment:
can we try to avoid this? tailwind doesn't like dynamic class names ([ref](https://tailwindcss.com/docs/content-configuration#dynamic-class-names))